src/python/gudhi/random_point_generators.cc

```diff
@@ -0,0 +1,68 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Hind Montassif
+ *
+ *    Copyright (C) 2021 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+#include <pybind11/pybind11.h>
+#include <pybind11/numpy.h>
+
+#include <gudhi/random_point_generators.h>
+
+#include <CGAL/Epick_d.h>
+
+namespace py = pybind11;
+
+
+typedef CGAL::Epick_d< CGAL::Dynamic_dimension_tag > Kern;
+
+template <typename Kernel>
+py::array_t<double> generate_points_on_sphere(py::object num_points, py::object dim, py::object radius) {
+    int npoints = num_points.cast<int>();
+    int d = dim.cast<int>();
+    double rad  = radius.cast<double>();
+    
+    py::gil_scoped_release release;
+
+    auto points_generated = Gudhi::generate_points_on_sphere_d<Kernel>(npoints, d, rad);
+    
+    py::gil_scoped_acquire acquire;
```

Comment:
"scoped" in the name means that the destructor automatically undoes what the constructor did, so you are not supposed to write the acquire part. Instead, you can add a scope to control when destruction happens
```c++
sometype points_generated;
{
  py::gil_scoped_release release;
  points_generated = function();
} // GIL reacquired here
```
You may want to move the definitions of points and buf before, so that the double loop copy can be in the nogil section. It is only python operations that need the GIL.

Reply:
Ok, done